sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -206,7 +206,8 @@ case class DropTableCommand(
     try {
       sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(tableName))
     } catch {
-      case _: NoSuchTableException if ifExists =>
+      case ae: AnalysisException
+        if ifExists && ae.cause.nonEmpty && ae.getCause.isInstanceOf[NoSuchTableException] =>
```

Comment:
can we follow https://github.com/apache/spark/pull/19713/files#diff-bc55b5f76add105ec32ae4107075b278R57

Reply:
```
scala> spark.version
res2: String = 2.3.0-SNAPSHOT

scala> sql("DROP TABLE IF EXISTS t")
res3: org.apache.spark.sql.DataFrame = []

scala> sql("DROP TABLE IF EXISTS t")
res4: org.apache.spark.sql.DataFrame = []
```

Unable to reproduce it.

Reply:
it's in the log, which means it's not a serious bug.

Reply:
I can see it in 2.2.1. It is a warning message.

Reply:
hmmm, it's fixed in master?

Reply:
@dongjoon-hyun can you verify it? thanks!

Reply:
I can reproduce it in master now.

Reply:
Yep. It exists in master, too.
I'll update like the example by @cloud-fan .